src/Resources/ResourceManager/Implementation/Deployments/NewAzureDeploymentWhatIfCmdlet.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Deployments
+{
+    using System;
+    using System.Management.Automation;
+    using Common.ArgumentCompleters;
+    using Management.ResourceManager.Models;
+    using SdkModels.Deployments;
+    using WindowsAzure.Commands.Utilities.Common;
+
+    /// <summary>
+    /// Creates a new deployment what-if.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New,
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeploymentWhatIf", SupportsShouldProcess = true,
```

Comment:
Does it change anything on Azure? I mean, if the cmdlet doesn't really do anything, `SupportsShouldProcess` is not needed.

Reply:
This will just invoke a background job in ARM and create an operation result for the specified scope. It doesn't touch any real resources though. Should that be considered as no change?

Reply:
Please remove it then. This cmdlet should be totally harmless from  user's perspective

Reply:
I tried to remove it but the static analysis failed. It seems for the verb New SupportsShouldProcess must be set to true...given that the name cannot be changed now, could we relax on this one?

Another reason why we may want to keep it: I just got some feedback from our PMs that we'll improve this cmdlet probably in next iteration so that it will ask users if they want to proceed and deploy the template. In that case, it will make real changes.

Reply:
The reason why you got that error was because we are running static checks, trying to make sure every PR follows our design guideline. And the checks are very strict -- so there are false positives sometimes.

The drawback of using `ShouldProcess` is you'll inevitably come to a situation:
`New-AzDeploymentWhatIf -WhatIf`
Just think about it...

Concerning the future plan you mentioned: I think it makes more sense to make that change to `New-AzDeployment` cmdlet. Consider this:
```powershell
> New-AzDeployment -Location *** -TemplateFile ***

The deployment will update the following scope:
.................

Are you sure you want to deploy the template
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"):
```

And user can skip it by specifying `-Confirm`.

What do you think?

Reply:
I am totally with you on the:`New-AzDeploymentWhatIf -WhatIf`, because it can confuse users a lot. I also think following the design guideline is important and we should remove `SupportsShouldProcess`. So is there a way to get around the static analysis error?

Speaking of the future plan, I like your idea. Actually we have some early discussion in our team. We’ll need to figure out if this would mean that deployment cmdlets do double duty of what-if and deployment or if we put that in the what-if cmdlets (same for CLI) and how to best fit it in the constraints imposed on us by Azure PS and CLI But that will to be decided after Ignite.

Reply:
Was able to find instructions: https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md

I'll follow it and suppress the error.

Reply:
Removed.